pkg/machines/components/storagePools/storagePool.jsx

```diff
@@ -29,68 +29,129 @@ import {
+        const name = (
+            <span id={`${idPrefix}-name`}>
+                { storagePool.name }
+            </span>);
+        const allocation = parseFloat(convertToUnit(storagePool.allocation, units.B, units.GiB).toFixed(2));
+        const capacity = parseFloat(convertToUnit(storagePool.capacity, units.B, units.GiB).toFixed(2));
+        const availableTooltipFunction = (max, now) => <Tooltip id='utilization-bar-tooltip-available'> Available {((max - now) / max).toFixed(2) * 100}% </Tooltip>;
+        const usedTooltipFunction = (max, now) => <Tooltip id='utilization-bar-tooltip-used'> Used {(now / max).toFixed(2) * 100}% </Tooltip>;
+        const size = (
+            <React.Fragment>
+                <UtilizationBar
+                    now={allocation}
+                    max={capacity}
+                    availableTooltipFunction={availableTooltipFunction}
+                    usedTooltipFunction={usedTooltipFunction}
+                />
+            </React.Fragment>
+        );
+        const sizeLabel = (
+            <React.Fragment>
+                {`${allocation} / ${capacity} GB`}
+            </React.Fragment>
+        );
+        const state = (
+            <React.Fragment>
+                { this.state.actionError && <span className='pficon-warning-triangle-o machines-status-alert' /> }
+                <span id={`${idPrefix}-state`}>
+                    { storagePool.active ? _("active") : _("inactive") }
+                </span>
+            </React.Fragment>);
+        const cols = [
+            {name, 'header': true},
+            size,
+            sizeLabel,
+            rephraseUI('connections', storagePool.connectionName),
+            state,
+        ];
+
+        const overviewTabName = (
+            <div id={`${idPrefix}-overview`}>
+                {_("Overview")}
+            </div>
+        );
+        const storageVolsTabName = (
+            <div id={`${idPrefix}-storage-volumes`}>
+                {_("Storage Volumes")}
+            </div>
+        );
+        let tabRenderers = [
+            {
+                name: overviewTabName,
+                renderer: StoragePoolOverviewTab,
+                data: {
+                    storagePool, actionError: this.state.actionError,
+                    actionErrorDetail: this.state.actionErrorDetail,
+                    onActionErrorDismiss: () => { this.setState({ actionError:  undefined }) }
+                }
+            },
+            {
+                name: storageVolsTabName,
+                renderer: StoragePoolVolumesTab,
+                data: { storagePool }
+            },
+        ];
+
+        return (
+            <ListingRow rowId={idPrefix}
+                columns={cols}
+                tabRenderers={tabRenderers}
+                listingActions={<StoragePoolActions actionErrorSet={this.actionErrorSet.bind(this)} storagePool={storagePool} />} />
+        );
+    }
+}
```

Comment:
Just a note and suggestion, these changes looks like refactor of the old code.  For review and git history it's better to split refactoring and new code into separate patches to make it clear which part is only refactoring.

Reply:
And the diff is so bad because I had to change this component from stateless to statefull to support the errors.

Reply:
Well :) changing component from stateless to statefull is the refactoring part and even though it's done in order to implement new functionality it should be in separate commit. It was just an suggestion based on good practices.